packages/@aws-cdk/assert/lib/assertions/have-output.ts

```diff
@@ -0,0 +1,82 @@
+import {Assertion} from "../assertion";
+import {StackInspector} from "../inspector";
+
+class HaveOutputAssertion extends Assertion<StackInspector> {
+  constructor(private readonly outputName?: string, private readonly exportName?: string, private outputValue?: any) {
+    super();
+  }
+
+  public get description(): string {
+    return `output\
+${this.outputName ? ' with name ' + this.outputName : ''}\
+${this.exportName ? ' with export name ' + this.exportName : ''}\
+${this.outputValue ? ' with value ' + JSON.stringify(this.outputValue) : ''}\
```

Comment:
The indentation here is all wrong. I'd rather we compose the description by just appending fragments to a `string` or something.